netlink-packet-route/src/rtnl/route/message.rs

```diff
@@ -18,7 +19,9 @@ impl Emitable for RouteMessage {
 
     fn emit(&self, buffer: &mut [u8]) {
         self.header.emit(buffer);
-        self.nlas.as_slice().emit(buffer);
+        self.nlas
+            .as_slice()
+            .emit(&mut buffer[self.header.buffer_len()..]);
```

Comment:
Doh... This proves I need more tests... I noticed this bug is in other message types as well.